pkg/ingestor/parser/csaf/parser_csaf.go

```diff
@@ -156,6 +191,17 @@ func findImpactStatement(tree *csaf.Vulnerability, product_id string) *string {
 	return nil
 }
 
+// findPkgSpec finds the package specification for the product with the
+// given ID in the CSAF document.
+//
+// It first calls findProductRef to get the product reference string for
+// the ID. If not found, returns an error.
+//
+// It then calls findPurl to get the purl for that product reference.
+// If not found, returns an error.
+//
+// Finally, it calls helpers.PurlToPkg to convert the purl to a package
+// spec to return.
```

Comment:
Thanks for adding the function comments. 

One note about this though.. the comments for the function should include what the function does (+ any implicit contracts with arguments, state and returns), but not so much as how the function does it - in this case calling a function is a means to do it, but what's really happening is business logic - i.e. someone should be able to implement a different version of the function that adhere's to the spec but uses different underlying code. 

In this case, just the first line should be sufficient

Reply:
I have shortened and edited the docstrings so that each maintains an overview of what the functions do.